include/deal.II/base/tensor.h

```diff
@@ -1235,8 +1233,9 @@ template <typename OtherNumber>
 DEAL_II_CONSTEXPR inline DEAL_II_ALWAYS_INLINE Tensor<rank_, dim, Number> &
 Tensor<rank_, dim, Number>::operator=(const Tensor<rank_, dim, OtherNumber> &t)
 {
-  if (dim > 0)
-    std::copy(&t.values[0], &t.values[0] + dim, &values[0]);
+  // std::copy is constexpr only from C++20 on.
+  for (unsigned int i = 0; i < dim; ++i)
+    values[i] = t.values[i];
```

Comment:
The comment is going to be hard to read without the context of this particular patch. The comment doesn't reference anything in the code, but rather something that's not there. I think you might want to say something like this: "The following loop could be written more concisely using `std::copy`, but that function is only `constexpr` from C++20 on."